src/transaction.js

```diff
@@ -250,8 +399,22 @@ Transaction.prototype.getId = function () {
   return [].reverse.call(this.getHash()).toString('hex')
 }
 
-Transaction.prototype.toBuffer = function () {
-  var buffer = new Buffer(this.byteLength())
+Transaction.prototype.toBuffer = function (withWitness) {
```

Comment:
I don't know if this is the right way to do it,  I've been playing with `toWitnessBuffer`,  or `WitnessTransaction` inheriting from `Transaction` where `toBuffer` is overwritten... still not sure.
I think we might just be better off detecting if a witness is needed and not bothering with a flag.